client/app/main/student/student.controller.js

```diff
@@ -2,9 +2,30 @@
 
 var app = angular.module('app');
 
-function StudentCtrl($scope, $state, $stateParams, Student, toastr) {
+function StudentCtrl($scope, $state, $stateParams, Student, 
+  AbsenceRecord, toastr) {
   $scope.student = Student.get({id: $stateParams.id});
 
+  $scope.calcPercentage = function() {
+    AbsenceRecord.current()
+      .$promise.then(function(records) {
+        var totPresent = [];
+        var totEnrolled = [];
+        var addToNext = function(prev, cur) {
+          return prev + cur;
+        };
+        records.forEach(function(record) {
+          totPresent.push(record.entries.present);
+          totEnrolled.push(record.entries.enrolled);
+        });
+        $scope.percentage = Math.floor(
+          (totPresent.reduce(addToNext) / totEnrolled.reduce(addToNext))
+          * 100);
+      });
+  };
+
+  $scope.calcPercentage();
+
```

Comment:
I'm trying to wrap my head around what this does. But it seems like you are calculating the percentage for all the students? 

The goal for this issue would be to display the percentage for the student that is being displayed in this detail view.